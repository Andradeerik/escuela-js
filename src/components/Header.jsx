import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import gravatar from '../utils/gravatar';
import '../assets/styles/components/Header.scss';
import logo from '../assets/static/logo-platzi-video-BW2.png';
import userIcon from '../assets/static/user-icon.png';

const Header = ({ user }) => (
  <header className="header">
    <Link to="/">
      <img className="header__img" src={logo} alt="Platzi Video" />
    </Link>
    {Object.keys(user).length > 0 && (
      <div className="header__menu">
        <div className="header__menu--profile">
          <img src={gravatar(user.email) || userIcon} alt={user.email} />
          <p>Perfil</p>
        </div>
        <ul>
          <li><a href="/">{user.name}</a></li>
          <li><a href="/">Cerrar Sesi√≥n</a></li>
        </ul>
      </div>
    )}
  </header>
);

const mapStateToProps = state => {
  return {
    user: state.user,
  };
};

export default connect(mapStateToProps, null)(Header);