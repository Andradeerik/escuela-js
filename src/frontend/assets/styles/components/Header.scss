.header {
  align-items: center;
  background-color: rgba(0,0,0,0.75);
  color: white;
  display: flex;
  height: 60px;
  justify-content: space-between;
  position: fixed;
  top: 0px;
  width: 100%;
  z-index: 3;
}

.header a {
  color: $color-green;
  text-decoration: none;
}

.header__item {
  align-items: center;
  display: flex;
  margin-left: 30px;
}

.header__item--title {
  color: $color-green;
  font-size: 30px;
}

.header__input-container {
  align-items: center;
  background-color: $gb-color;
  border: 1px solid white;
  display: flex;
  margin-right: 30px;
  padding: 3px;
}

.header__input-container--text {
  background-color: $gb-color;
  border: none;
  color: white;
  font-size: 13px;
  height: 30px;
  outline: none;
  padding-left: 5px;
}
