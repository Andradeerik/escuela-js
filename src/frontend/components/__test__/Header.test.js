import React from 'react';
import { create } from 'react-test-renderer';
import Header from '../Layout';

describe('Header component', () => {
  test('Matches the snapshot', () => {
    const header = create(<Header />);
    expect(header.toJSON()).toMatchSnapshot();
  });
});
