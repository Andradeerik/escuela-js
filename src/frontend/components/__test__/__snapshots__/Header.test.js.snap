// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Matches the snapshot 1`] = `
<div
  className="App"
>
  <footer
    className="footer"
  >
    <a
      href="/"
    >
      Terminos de uso
    </a>
    <a
      href="/"
    >
      Declaraci√≥n de privacidad
    </a>
    <a
      href="/"
    >
      Centro de ayuda
    </a>
  </footer>
</div>
`;
